.github/workflows/tests.yml

```diff
@@ -120,42 +145,62 @@ jobs:
           pip install --upgrade coveralls pytest-cov pytest
           pip list
 
-      - name: Set environment variables
-        shell: bash
+      - name: Set environment variables (Linux)
         run: |
-          echo "LD_LIBRARY_PATH=$GEOS_INSTALL/lib" >> $GITHUB_ENV
-          echo $GEOS_INSTALL/bin >> $GITHUB_PATH
+          echo "${{ env.GEOS_INSTALL }}/bin" >> $GITHUB_PATH
+          echo "LD_LIBRARY_PATH=${{ env.GEOS_INSTALL }}/lib" >> $GITHUB_ENV
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+
+      - name: Set environment variables (macOS)
+        run: |
+          echo "${{ env.GEOS_INSTALL }}/bin" >> $GITHUB_PATH
+          echo "DYLD_LIBRARY_PATH=${{ env.GEOS_INSTALL }}/lib" >> $GITHUB_ENV
+          echo "LDFLAGS=-Wl,-rpath,${{ env.GEOS_INSTALL }}/lib" >> $GITHUB_ENV
+          install_name_tool -add_rpath @executable_path/. ${{ env.GEOS_INSTALL }}/lib/libgeos_c.dylib
+          otool -l ${{ env.GEOS_INSTALL }}/lib/libgeos_c.dylib
+        if: ${{ matrix.os == 'macos-latest' }}
+
+      # Windows requires special treatment:
+      # - geos-config does not exist, so we specify include and library paths
+      # - Python >=3.8 ignores the PATH for finding DLLs, so we copy them into the package
+      - name: Set environment variables + copy DLLs (Windows)
+        run: |
+          echo "${{ env.GEOS_INSTALL }}\bin" >> $GITHUB_PATH
+          if [ "${{ matrix.architecture }}" == "x86" ]; then
+            DLL_DIR=DLLs_x86_VC9
+          elif [ "${{ matrix.architecture }}" == "x64" ]; then
+            DLL_DIR=DLLs_AMD64_VC9
+          fi
+          cp geosinstall/geos-${{ matrix.geos }}/bin/*.dll $DLL_DIR
+          echo 'GEOS_LIBRARY_PATH=${{ env.GEOS_INSTALL }}\bin\geos_c.dll' >> $GITHUB_ENV
+          echo 'GEOS_INCLUDE_PATH=${{ env.GEOS_INSTALL }}\include' >> $GITHUB_ENV
+        if: ${{ matrix.os == 'windows-2019' }}
 
       - name: Build and install Shapely
-        shell: bash
         run: |
           pip install -v -e .
 
       - name: Overview of the Python environment (pip list)
-        shell: bash
         run: |
           pip list
 
       - name: Run tests
-        shell: bash
-        continue-on-error: ${{ matrix.geos == 'main' }}
         run: |
+          export DYLD_LIBRARY_PATH=${{ env.DYLD_LIBRARY_PATH }}
```

Comment:
this appears to be necessary! Despite being part of `$GITHUB_ENV`, it cannot be seen i.e. `echo "DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH"` shows a blank entry.